From: Harry Kim <harry.kim@newracom.com>
Subject: [PATCH] make_ext4fs: mark inode bitmap padding as used

Bits beyond s_inodes_per_group are padding rather than free inodes and must
be set in every inode bitmap.  Leaving the calloc() initialized zeroes in
that range makes e2fsck report a filesystem error after image creation.

Signed-off-by: Harry Kim <harry.kim@newracom.com>

--- a/allocate.c
+++ b/allocate.c
@@ -282,6 +282,7 @@ void reduce_allocation(struct block_allocation *alloc, u32 len)
 static void init_bg(struct block_group_info *bg, unsigned int i)
 {
 	int header_blocks = 2 + aux_info.inode_table_blocks;
+	u32 inode;
 
 	bg->has_superblock = ext4_bg_has_super_block(i);
 
@@ -293,6 +294,9 @@ static void init_bg(struct block_group_info *bg, unsigned int i)
 	bg->block_bitmap = bg->bitmaps;
 	bg->inode_bitmap = bg->bitmaps + info.block_size;
 
+	for (inode = info.inodes_per_group; inode < info.block_size * 8; inode++)
+		bitmap_set_bit(bg->inode_bitmap, inode);
+
 	bg->header_blocks = header_blocks;
 	bg->first_block = aux_info.first_data_block + i * info.blocks_per_group;
 
